src/metadata.rs

```diff
@@ -50,13 +46,52 @@ impl From<io::Error> for MetadataError {
     }
 }
 
-fn get_date_time_created(path: &Path) -> Result<chrono::DateTime<FixedOffset>, MetadataError> {
-    Ok(FixedOffset::west(0).timestamp(fs::metadata(path)?.ctime(), 0))
+fn result_to_option<T, E: Error>(res: Result<T, E>, checkpoint: String) -> Option<T> {
+    match res {
+        Ok(t) => Some(t),
+        Err(e) => {
+            eprint!("Error occurred during {}, {}", checkpoint, e);
+            None
+        }
+    }
 }
 
-fn convert_exif_date_time_to_chrono_date_time_fixed_offset(
+fn open_file(path: &Path, file_name: &String) -> Option<fs::File> {
```

Comment:
I think we should use `Result` here as the return value because it already encodes the concept of an error case by design (same way you wouldn't return an Optional in Java instead of throwing an exception).  By extension, I'd make `read_metadata()` return a `Result`.

`Result` also allows using `?` which a nice way to chain function calls that return errors.